scripts/setup-ubuntu.sh

```diff
@@ -52,13 +52,15 @@ function install_build_prerequisites {
   ${SUDO} apt install -y libunwind-dev
   ${SUDO} apt install -y \
     build-essential \
-    cmake \
+    python3-pip \
     ccache \
     curl \
     ninja-build \
     checkinstall \
     git \
     wget
+
+    ${SUDO} pip3 install cmake==3.28.3
```

Comment:
I think we should use it like this, as using pip with sudo can break stuff:
```suggestion
    pip3 install --user cmake==3.28.3
```

Reply:
Is there a reason to pin at 3.28.3? The latest build is 3.29.3. If you’re pinning to a specific version for reproducibility it might as well start at the latest version unless there’s a reason to do otherwise.

Reply:
Proposing:
```suggestion
    pip3 install cmake==3.29.3
```

(Feel free to merge this without any further feedback from me — my response times will be delayed for a while.)

Reply:
I think that would also be fine, I don't have a strong opinion.

Reply:
though the newer version _shouldn't_ change anything as it will act like 3.28.3 through policies. Actually on second thought I think it's better to use the actual minimal version so any accidental use of unsupported features shows up in CI.

Reply:
@bdice I preferred to stick to cmake version 3.28 because it is the same version that Ubuntu 24.04 uses, which means that developers of Ubuntu know that it is safe to use. I don't oppose to using cmake 3.29 but I did not see specific features that may benefit Velox, and I don't know whether it introduces breaking changes or not.

Reply:
@bdice Thanks for the feedback, I'm addressing your comment right now.

Reply:
Matching Ubuntu 24.04 is fine with me! It’s okay to pin to 3.28.3. I just wasn’t aware of the rationale. I’ll let you decide how to proceed. 👍